subworkflows/local/align_MT.nf

```diff
@@ -0,0 +1,72 @@
+include { GATK4_MERGEBAMALIGNMENT as GATK4_MERGEBAMALIGNMENT_MT             } from '../../modules/nf-core/modules/gatk4/mergebamalignment/main'
+include { PICARD_ADDORREPLACEREADGROUPS as PICARD_ADDORREPLACEREADGROUPS_MT } from '../../modules/nf-core/modules/picard/addorreplacereadgroups/main'
+include { PICARD_MARKDUPLICATES as PICARD_MARKDUPLICATES_MT                 } from '../../modules/nf-core/modules/picard/markduplicates/main'
+include { SAMTOOLS_INDEX as SAMTOOLS_INDEX_MT                               } from '../../modules/nf-core/modules/samtools/index/main'
+include { SAMTOOLS_SORT as SAMTOOLS_SORT_MT                                 } from '../../modules/nf-core/modules/samtools/sort/main'
+include { HAPLOCHECK as HAPLOCHECK_MT                                       } from '../../modules/nf-core/modules/haplocheck/main'
+include { GATK4_MUTECT2 as GATK4_MUTECT2_MT                                 } from '../../modules/nf-core/modules/gatk4/mutect2/main'
+
+workflow ALIGN_MT {
+    take:
+        fastq         // channel: [ val(meta), path('*.fastq.gz') ]
+        ubam          // channel: [ val(meta), path('*.bam') ]
+        index         // channel: [ /path/to/bwamem2/index/ ]
+        fasta         // channel: [ genome.fasta ]
+        dict          // channel: [ genome.dict ]
+        fai           // channel: [ genome.fai ]
+        intervals_mt  // channel: [ file(non_control_region.chrM.interval_list) ]
+
+    main:
+        ch_versions = Channel.empty()
+
+        // Outputs bam files
+        BWAMEM2_MEM_MT ( fastq , index, true)
+        ch_versions    = ch_versions.mix(BWAMEM2_MEM_MT.out.versions.first())
+        ch_mt_bam      =  BWAMEM2_MEM_MT.out.bam
+        ch_fastq_ubam  = ch_mt_bam.join(ubam, by: [0])
+
+        // Merges bam files
+        GATK4_MERGEBAMALIGNMENT_MT ( ch_fastq_ubam, fasta, dict )
+        ch_versions = ch_versions.mix(GATK4_MERGEBAMALIGNMENT_MT.out.versions.first())
+
+        // Add read group to merged bam file
+        PICARD_ADDORREPLACEREADGROUPS_MT ( GATK4_MERGEBAMALIGNMENT_MT.out.bam )
+        ch_versions = ch_versions.mix(PICARD_ADDORREPLACEREADGROUPS_MT.out.versions.first())
+
+        // Marks duplicates
+        PICARD_MARKDUPLICATES_MT ( PICARD_ADDORREPLACEREADGROUPS_MT.out.bam )
+        ch_versions = ch_versions.mix(PICARD_MARKDUPLICATES_MT.out.versions.first())
+
+        // Sort bam file
+        SAMTOOLS_SORT_MT (PICARD_MARKDUPLICATES_MT.out.bam)
+        ch_versions = ch_versions.mix(SAMTOOLS_SORT_MT.out.versions.first())
+
+        // Index bam file
+        SAMTOOLS_INDEX_MT(SAMTOOLS_SORT_MT.out.bam)
+        ch2=SAMTOOLS_SORT_MT.out.bam.join(SAMTOOLS_INDEX_MT.out.bai, by: [0])
+        ch3=ch2.combine(intervals_mt)
+        ch_versions = ch_versions.mix(SAMTOOLS_INDEX_MT.out.versions.first())
+
+        // Calls variants with Mutect2
+        GATK4_MUTECT2_MT ( ch3, fasta, fai, dict, [], [], [], [] )
+        ch_versions = ch_versions.mix(GATK4_MUTECT2_MT.out.versions.first())
```

Comment:
Maybe this should be part of a new sub workflow called `CALL_MT_VARIANTS` or `CALL_AND_ANNOTATE_MT_VARIANTS` depending on how much more that goes into that particular sub workflow. It's not really alignment right

Reply:
That would be problematic because the variants will need to be called on both sides of the "branches" with and without the shift. Maybe I can change the name of the sub workflow and call it align_and_call_MT?